File: C/pset1/problems-2020-x-credit/credit.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    long number = 0;
    //Memaksa User Masukkan digit positif
    do
    {
        number = get_long("Number: ");
    }
    while (number < 0);

    //Declare data types
    long digita = 0;
    long digitb = 0;
    long suma = 0;
    long sumb = 0;
    int counter = 0;
    //sum digit genap dari belakang lalu kali 2, count digit
    for (long x = 100; ((number * 10) / x) > 1 ; x = x * 100)
    {
        digita = ((number % x) / (x / 10)) * 2;
        //bila 2 digit, pisahkan digit lalu tambahkan
        if (digita > 9)
        {
            long digit1 = digita % 10;
            long digit2 = digita / 10;
            suma = suma + digit1 + digit2;
            counter++;
        }
        //bila 1 digit, langsung sum
        else
        {
            suma = suma + digita;
            counter++;
        }
    }

    //sum ganjil dari belakang, count digit
    for (long x = 10; ((number * 10) / x) > 1 ; x = x * 100)
    {
        digitb = ((number % x) / (x / 10));
        sumb = sumb + digitb;
        counter++;
    }
    
    //checksum remainder
    int checksum = (suma + sumb) % 10;
    
    //membuat pangkat dari counter
    long power = pow(10, counter);
    //untuk 1 digit depan
    long fd = (number % power) / (power / 10);
    //untuk 2 digit depan
    long sd = (number / (power / 100));
    
    //condition tiap kartu
    if (checksum == 0 && (counter == 16 || counter == 13) && fd == 4)
    {
        printf("VISA\n");
    }
    else if (checksum == 0 && counter == 16 && sd > 50 && sd < 56)
    {
        printf("MASTERCARD\n");
    }
    else if (checksum == 0 && counter == 15 && (sd == 34 || sd == 37))
    {
        printf("AMEX\n");
    }
    else
    {
        printf("INVALID\n");
    }
}

File: Python, SQL, HTML, CSS, Flask, Javascript/pset6/problems-2020-x-sentimental-credit/credit.py
from cs50 import get_string

# Get user input
n = get_string("Number: ")
while n.isnumeric() == False:
    n = get_string("Number: ")

# Count Second to Last Digit to First
counter = 2
sum1 = 0
while len(n)/counter >= 1:
    digit = str(2 * int(n[len(n) - counter]))
    if len(digit) == 2:
        sum1 = sum1 + int(digit[0]) + int(digit[1])
        counter = counter + 2
    elif len(digit) == 1:
        sum1 = sum1 + int(digit)
        counter = counter + 2

# Count First to Last Digit to First
counter = 1
sum2 = 0
while len(n)/counter >= 1:
    sum2 = sum2 + int(n[len(n) - counter])
    counter = counter + 2

# Condition for Card Number
if (sum1+sum2) % 10 != 0:
    print("INVALID")

elif len(n) == 15 and (n[:2] == "37" or n[:2] == "34"):
    print("AMEX")

elif len(n) == 16 and (int(n[:2]) > 50 and int(n[:2]) < 56):
    print("MASTERCARD")

elif (len(n) == 16 or len(n) == 13) and (n[:1] == "4"):
    print("VISA")